src/compile/data/impute.ts

```diff
@@ -1,32 +1,84 @@
-import {ImputeTransform} from '../../transform';
+import {normalizeimputeProperties} from '../../impute';
+import {ImputeSequence, ImputeTransform, isImputeSequence} from '../../transform';
 import {duplicate} from '../../util';
-import {VgImputeTransform} from '../../vega.schema';
+import {VgFormulaTransform, VgImputeTransform, VgSignalRef, VgWindowTransform} from '../../vega.schema';
+import {UnitModel} from '../unit';
 import {DataFlowNode} from './dataflow';
 
+// TODO Rename to ImputeNode
 export class ImputeTransformNode extends DataFlowNode {
   public clone() {
     return new ImputeTransformNode(this.parent, duplicate(this.transform));
   }
 
   public producedFields() {
     // typescript detects true as boolean type
-    return {[this.transform.impute]: true as true} ;
+    return {[this.transform.impute]: true as true};
   }
 
   constructor(parent: DataFlowNode, private transform: ImputeTransform) {
     super(parent);
+
+  }
+
+  private processSequence(keyvals: ImputeSequence): VgSignalRef {
+    const {start, stop, step=1} = keyvals;
+    return {signal: `sequence(${start}, ${stop}, ${step})`};
   }
 
-  public assemble(): VgImputeTransform {
-    const {impute, key, keyvals, method, groupby, value} = this.transform;
-    return {
-      type: 'impute',
-      field: impute,
-      key,
-      ...(keyvals? {keyvals}: {}),
-      ...(method? {method}: {}),
-      ...(groupby? {groupby}: {}),
-      ...(value? {value}: {})
-    };
+  public static makeFromTransform(parent: DataFlowNode, imputeTransform: ImputeTransform): ImputeTransformNode {
+    return new ImputeTransformNode (parent, imputeTransform);
+  }
+
+  public static makeFromEncoding(parent: DataFlowNode, model: UnitModel) {
+    const transformRepresentation = normalizeimputeProperties(model.encoding);
```

Comment:
`normalizeimputeProperties`  is only used here.

Since this internal logic of a compiler, let's remove the `normalizeimputeProperties` method as just move the logic here.